tests/playbooks/down_profile.yml

```diff
@@ -8,3 +8,12 @@
         state: down
   roles:
     - linux-system-roles.network
+
+- name: Delete the interface when the network provider is initscripts
+  hosts: all
+  tasks:
+    - name: Delete the interface when the network provider is initscripts
+      command: ip link del "{{ profile }}"
+      ignore_errors: true   # noqa ignore-errors
+      when:
+        - network_provider == "initscripts"
```

Comment:
The other change implies that the goal is to support switching from initscripts to NM (which seems to be correct to me). This means, that the tests should not be altered and that we should have an actual test that tests that switching from initscripts to NM actually works.

Reply:
I have explained this in the reason of this change in the first comment: The absent and down action of initscript provider will not remove the bridge interface which fail the assertion in `tests_bridge_nm.yml`.

i will include a dedicate test case for this.

Reply:
Dedicate test case included and reproduced on RHEL/CentOS 7.

Reply:
In the new integration test you use a task in a file `delete_interface.yml` which removed interfaces and is slightly different (using the interface variable and also setting `changed_when`):

https://github.com/linux-system-roles/network/blob/main/tests/tasks/delete_interface.yml

Also, in the new integration test, you are writing down the content of the original `down_profile.yml` instead of using it, since it is now not down_profile but down_profile+delete_interface. To make it easier to understand what is going on, it seems to me that a file that is named so that it explains what happens, would be better.

Reply:
this does not seem to be resolved. At least the `changed_when` should be added and the other remarks should be considered.

Reply:
I have introduced a new patch addressing your comments. Could you check if it looks good to you know? Thanks!